packages/core/src/lib/jdk/KeyTool.ts

```diff
@@ -77,4 +85,65 @@ export class KeyTool {
     await execute(keytoolCmd, env);
     this.log.info('Signing Key created successfully');
   }
+
+  /**
+   * Runs `keytool --list` on the keystore / alias provided on the {@link KeyOptions}.
+   *
+   * @param {KeyOptions} keyOptions parameters for they key to be listed.
+   * @returns {Promise<string>} the raw output of the `keytool --list` command
+   */
+  async list(keyOptions: KeyOptions): Promise<string> {
+    if (!existsSync(keyOptions.path)) {
+      throw new Error(`Couldn't find signing key at "${keyOptions.path}"`);
+    }
+    const keyListCmd = [
+      'keytool',
+      '-list',
+      '-v',
+      `-keystore \"${keyOptions.path}\"`,
+      `-alias \"${keyOptions.alias}\"`,
+      `-storepass \"${keyOptions.password}\"`,
+      `-keypass \"${keyOptions.keypassword}\"`,
+    ];
+    const env = this.jdkHelper.getEnv();
+    const result = await execute(keyListCmd, env);
+    return result.stdout;
+  }
+
+  /**
+   * Runs `keytool --list` on the keystore / alias provided on the {@link KeyOptions}. Currently,
+   * only extracting fingerprints is implemented.
+   *
+   * @param {KeyOptions} keyOptions parameters for they key to be listed.
+   * @returns {Promise<KeyInfo>} the parsed output of the `keytool --list` command
+   */
+  async keyInfo(keyOptions: KeyOptions): Promise<KeyInfo> {
+    const rawKeyInfo = await this.list(keyOptions);
+    return KeyTool.parseKeyInfo(rawKeyInfo);
+  }
+
+  /**
+   * Parses the output of `keytool --list` and returns a structured {@link KeyInfo}. Currently,
+   * only extracts the fingerprints.
+   */
+  static parseKeyInfo(rawKeyInfo: string): KeyInfo {
+    const lines = rawKeyInfo.split('\n');
+    const fingerprints: Map<string, string> = new Map();
+    const fingerprintTags = ['SHA1', 'SHA256'];
+
+    lines.forEach((line) => {
+      line = line.trim();
+      fingerprintTags.forEach((tag) => {
+        if (line.startsWith(tag)) {
+          // a fingerprint line has the format <tag>: <value>. So, we account for the extra colon
+          // when substringing and then trim to remove whitespaces.
+          const value = line.substring(tag.length + 1, line.length).trim();
+          fingerprints.set(tag, value);
+        }
+      });
```

Comment:
This could just be:

    if (line.startsWith("SHA")) {
        const parts = str.split(":")
        fingerprints.set(parts[0], parts[1].trim());
    }

Although I think your version is probably nicer.